### PR TITLE
Fix queue scalability problems

### PR DESCRIPTION
**Pull request by DanielCosme**
This PR adds the option of configuring Temporal's worker capacity to run concurrent workflows, mainly via the: `MaxConcurrentWorkflowTaskExecutionSize` and `MaxConcurrentSessionExecutionSize` options.

The problems is that when queuing 1000+ SIPs into Enduro, the system would fundamentally start failing workflows via Heartbeat Timeouts deadlines being exceeded. Initially I assumed that setting up the timeout values higher (around 1h) the issue would be solved, but then at 4000+ SIPs in queue, the timeouts would re-appear. Then I realised that what Enduro considers a "queue" was not being really queued, as far as Temporal was concerned. 4000k SIPs queued would actually be Executing workflows and as this number increases so it would the resources used by Temporal. A Temporal queue is supposed to be cheap to maintain, so something was wrong. 

How an actual queue in Temporal would look like (for Enduro's use case) is with the workflow running but with the `internalSessionCreation` activity in a `Scheduled` state but nothing else, this is very cheap; as opposed as having the workflow assigned to the worker and being in an execution state (which at a higher number takes considerable resources).

So by limiting these two configuration options all the scaling and performance related issues with the queue simply dissapeared, up to 30k Scheduled SIPs (I did not a bigger queue than that). 

Also, initially I tried configuring only the `MaxConcurrentSessionExecutionSize` alone and the timeout issues were no more. However, at 400+ SIPs in queue Enduro would stop sending SIPs to AM and the entire pipeline will grind to a halt with all the system fundamentally in iddle, not sure why this happened. Then I applied the same value to `MaxConcurrentWorkflowTaskExecutionSize`, after that the system started running very smoothly.

Finally, after running a lot tests I would recommend that the value for `MaxConcurrentTransfers` would be equal to the total sum of the `Capacity` value for each AM pipeline. In other words it is my opinion that this value should be directly proportional to the total AM pipeline capacity.

Also, there was one heartbeat timeout value that was still hard-coded, I added a change to use the value from the configuration.

### CODE REVIEW

**Review by sevein — APPROVED** [review]
Thank you, Daniel! I'd like to suggest reconsidering the configuration name `MaxConcurrentTransfers`. We use workflows for other purposes, e.g. batch worfklows or bulk workflows. Mapping the worker parameters `MaxConcurrentWorkflowTaskExecutionSize` and `MaxConcurrentSessionExecutionSize` directly to the configuration might be the most flexible approach for now.

**Comment by scollazo** [comment]
Could you create a new release once this is merged?

**Comment by DanielCosme** [comment]
@scollazo I don't know how to create releases here.

**Comment by DanielCosme** [comment]
@sevein Like this? 15253627578539fbb7353e6b32f2b988ca924760

**Comment by sevein** [comment]
Works for me. Does it matter that they start with a capital letter?

**Comment by DanielCosme** [comment]
@sevein I wonder if instead of exposing Temporal's internal configuration options we just set those concurrent workflows/sessions values directly from the total AM pipeline capacity. Getting rid of the need for configuring values that require understanding the internals of the workflow engine. Doing this is fundamentally what I found fixed the issue, making these values directly proportional to the pipeline capacity. What do  you think?

**Comment by sevein** [comment]
> @sevein I wonder if instead of exposing Temporal's internal configuration options we just set those concurrent workflows/sessions values directly from the total AM pipeline capacity. Getting rid of the need for configuring values that require understanding the internals of the workflow engine. Doing this is fundamentally what I found fixed the issue, making these values directly proportional to the pipeline capacity. What do you think?

That's assuming that `MaxConcurrentWorkflowTaskExecutionSize` is bound only to pipeline capacity but that's not completely accurate, we'd still have to acommodate tasks coming from batch and bulk workflow executions. You could compute the defaults dynamically based on that criteria or similar, but I think it may still be worth making these config attributes available to the user as we explore them further?

I'm not particularly worried about too many config attributes, maybe because it's still v0.y.z.

**Comment by DanielCosme** [comment]
> > @sevein I wonder if instead of exposing Temporal's internal configuration options we just set those concurrent workflows/sessions values directly from the total AM pipeline capacity. Getting rid of the need for configuring values that require understanding the internals of the workflow engine. Doing this is fundamentally what I found fixed the issue, making these values directly proportional to the pipeline capacity. What do you think?
> 
> That's assuming that `MaxConcurrentWorkflowTaskExecutionSize` is bound only to pipeline capacity but that's not completely accurate, we'd still have to acommodate tasks coming from batch and bulk workflow executions. You could compute the defaults dynamically based on that criteria or similar, but I think it may still be worth making these config attributes available to the user as we explore them further?
> 
> I'm not particularly worried about too many config attributes, maybe because it's still v0.y.z.

@sevein 

Ok, that makes sense, thank you. 

About the documentation, I created this commit for it. what do you think? 

aeb4cdb0c733226d621dfc8013d3d298efb6d641

File: website/content/en/docs/user-manual/configuration.md [hunk 1]
```diff
@@ -462,6 +462,51 @@ does not include a document with checksums, e.g. `checksum.sha1`.
...
+The string should be constructed as a sequence of decimal numbers, each with
+optional fraction and a unit suffix, such as "30m", "24h" or "2h30m".
+Valid time units are "ns", "us" (or "µs"), "ms", "s", "m", "h".
+
+E.g.: `5m` (String)
+
+#### `maxConcurrentWorkflowsExecutionsSize` (int)
+
+Sets the maximum number of concurrent workflow executions that Enduro will accept from the workflow engine.
+A good rule of thumb is to make this value 2 times the sum total of the Archivematica pipeline capacity.
```
Comment (sevein) [inline website/content/en/docs/user-manual/configuration.md:481]:
```suggestion
A good rule of thumb is to set this value to twice the sum of the capacities of all the configured pipelines. For example, if two pipelines are configured with a capacity of `5` each, the value should be `20`.
```

**Review by sevein — APPROVED** [review]
LGTM. Please remember to update the markdown file by wrapping all lines to 80 columns.

**Pull request merged**